Adding clean-architecture

AmberSimsDev opened the pull request:
I made the following changes (as instructed)
------------------------------------------------
* Created new branch from Main 
  * New Branch called 'clean-architecture'
* Created 'domain' package

Regarding Architecture:
---------------------------------
* Renamed models to '(insertname)Dto'
* Moved dtos into data - > model directory

Note: App still runs with no issues (/^ 0^)/ : .

AmberSimsDev committed:
work in progress

AmberSimsDev committed:
Update Project_Default.xml

Created directory 'remote'
Created subfolder 'api'
Created subfolder 'model'
Moved PokemonApiService - > 'api'
Moved PokemonSpriteService -> 'api'
Moved PokeBerry  -> 'model'
Created PokemonDataLayerDTO
Moved PokemonDataLayerDTO -> 'model'

AmberSimsDev committed:
Update PokemonDataLayerDTO.kt

Created BerryDetailDTO
Created BerryDTO
Completed getBerryDetailDTOList
Completed getBerryDTOList
Fixed  data class BerryDetails
Omitted fun getFirmness

Prometheus3k commented on app/src/main/java/asdigital/myberryapp/data/remote/model/PokeBerry.kt:
```diff
@@ -1,9 +1,9 @@
 
 import com.google.gson.annotations.SerializedName
 
 data class Berry(
```
data.remote.model contains all the model objects from the REST API. These are DTOs. So:
1) End the name in DTO
2) The data class name should match the filename. 

So either data class BerryDTO in file BerryDTO.kt or PokeBerryDTO in file PokeBerryDTO.kt

Prometheus3k commented on app/src/main/java/asdigital/myberryapp/data/remote/model/BerryDetailDTO.kt:
```diff
@@ -0,0 +1,89 @@
+    val name: String,
+    @SerializedName("growth_time") val growthTime: Int,
+    @SerializedName("max_harvest") val maxHarvest: Int,
+    @SerializedName("natural_gift_power") val naturalGiftPower: Int,
```
it's more common to format as follows
```
    @SerializedName("growth_time") 
    val growthTime: Int,
    @SerializedName("max_harvest")
    val maxHarvest: Int,
    @SerializedName("natural_gift_power") 
    val naturalGiftPower: Int,
```
Check @SerlizedName below as well.

Prometheus3k commented on app/src/main/java/asdigital/myberryapp/data/remote/api/PokemonApiService.kt:
```diff
@@ -1,13 +1,11 @@
 import retrofit2.converter.gson.GsonConverterFactory
 import retrofit2.http.GET
 import retrofit2.http.Path
-import retrofit2.http.Url
```
Your GETs should be returning a DTO class. BerryResponseDTO / BerryDetailDTO

Prometheus3k commented on app/src/main/java/asdigital/myberryapp/data/remote/api/PokemonSpritesApiService.kt:
```diff
@@ -1,4 +1,4 @@
-package asdigital.myberryapp.data.remote
+package asdigital.myberryapp.data.remote.api
 
 fun getBerrySpriteUrl(name: String): String {
```
remote.api files should define a fetch from some REST Api, like PokemonApiService.
This is just a function that takes name and returns a URL. So not an api service (it doesn't GET anything). 
To me this is a PokemonSpriteUrlMapper.

Prometheus3k commented on app/src/main/java/asdigital/myberryapp/data/remote/model/PokemonDataLayerDTO.kt:
```diff
@@ -0,0 +1,89 @@
+
+//HERE WE HAVE OUR GETTERS
+
+fun getBerryDetailDTOList(berryDetail: List<BerryDetail>): List<BerryDetailDTO> {
```
This doesn't seem correct. Your ApiService returns a BerryDetail and now you map it a BerryDetailDTO. 

DTO is the beginning of everything. Your APIService returns DTO and you then map to Domain models. If you cmd+click this function name you'll see it's not used.

<img width="768" height="123" alt="Screenshot 2025-09-04 at 01 30 53" src="https://github.com/user-attachments/assets/40f4e41c-fcb7-4db8-8f3f-7d4eb0e687ee" />

Prometheus3k commented on app/src/main/java/asdigital/myberryapp/data/remote/model/PokemonDataLayerDTO.kt:
```diff
@@ -0,0 +1,89 @@
+    }
+}
+
+fun getBerryDTOList(results: List<Berry>): List<BerryDTO> {
```
same problem as above. Unused so remove.

Prometheus3k commented on app/src/main/java/asdigital/myberryapp/data/remote/model/BerryDetailDTO.kt:
```diff
@@ -0,0 +1,89 @@
+data class NaturalGiftType(
+    val name: String,
+    val url: String
+)
```
These are unused. It looks like a duplicate of PokeBerry.kt

remove all the unused data classes. to check if it's used, ctrl+click the class name

Prometheus3k commented on app/src/main/java/asdigital/myberryapp/data/remote/model/BerryDetailDTO.kt:
```diff
@@ -0,0 +1,89 @@
+//data class BerryResponse( val results: List<Berry>)
+//data class BerryDetailResponse(val berryDetail: List<BerryDetail>)
+
+data class BerryDetailDTO(
```
Since your APIService is not using any DTO classes, nothing in this file is actually being used. 
If you comment out the whole thing, you'll see no errors; an indication that it's not being used. 
APIService should be using these DTOs

Prometheus3k reviewed (changes requested):
Not bad start but we need to clean up this clean arch first step. 
I think you've left a lot of unused dead code. 
1. delete PokemonDataLayerDTO, no one is using it.
2. PokemonApiService is using the classes in PokeBerry.kt. These are your DTOs and name should be ending in DTO. BerryDetailResponse is unused in there so can remove.
3. Mistake in the data package. you've inverted api.remote
<img width="375" height="220" alt="Screenshot 2025-09-04 at 01 49 34" src="https://github.com/user-attachments/assets/51d8a1b2-dcb6-4ca3-8dd4-57d197a85fa8" />
It should be: 
data.api.remote
data.model

We need to get your DTOs and packages right before we can move onto mappers and domain

AmberSimsDev committed:
DTOs Added

Created Model Package  in  Remote package
Created Repositiories:
-BerryDTO
-BerryDetailDTO
-Berry ResponseDTO
-BerryDetailResponseDTO

Refactored objects w/o DTO added in
-PokemonApiService
-PokeScreen
-BerryDetailViewModel

AmberSimsDev commented:
Okay I made new commits, taking into account your comments ( ^^')
